### PR TITLE
nest: migrate to homebrew/core

### PR DESCRIPTION
**Pull request by tammoippen**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Reopening #21561, as the upstream PR (https://github.com/nest/nest-simulator/pull/872) is now merged. (Sorry for the inconvenience of another PR - I messed up my fork.)

### CODE REVIEW

**Comment by tammoippen** [comment]
Failure is because of `brew audit nest --online --new-formula`, as `patches` are required for building nest 2.14.0, as discussed here https://github.com/Homebrew/homebrew-core/pull/21561#issuecomment-351337535.

**Comment by tammoippen** [comment]
Is there something blocking this PR?

File: Formula/nest.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
+class Nest < Formula
+  desc "The Neural Simulation Tool (NEST) with Python2 bindings (PyNEST)"
+  homepage "http://www.nest-simulator.org/"
+  url "https://github.com/nest/nest-simulator/archive/v2.14.0.tar.gz"
+  sha256 "afaf7d53c2d5305fac1257759cc0ea6d62c3cebf7d5cc4a07d4739af4dbb9caf"
+
+  bottle do
```
Comment (fxcoudert) [inline Formula/nest.rb]: No bottle block

File: Formula/nest.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    sha256 "4f7d765cd75ae3272ce15bb293c20a6cdfc9ccb551f0a07d00016c75ae0ef267" => :high_sierra
+    sha256 "60e4b0595c5d11003ffe1eca3fdca3c197b8711336ead21eba67620db96454c0" => :sierra
+    sha256 "8a4c8e6d0d2e4ce93d76af73197c693f8c303095800e82b7266b922848f1a39b" => :el_capitan
+    sha256 "6207dea13f90b17a95313b964323f7971cd5e2507a38b0ffd5d64b2f346e7851" => :x86_64_linux
+  end
+
+  # Add patches for matplotlib removal in upstream:
+  patch do
```
Comment (fxcoudert) [inline Formula/nest.rb]: 😢 patches… can upstream make a new release?

File: Formula/nest.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    url "https://github.com/nest/nest-simulator/commit/d60e189123b994764bef8179aa91b4b001dad1dd.patch?full_index=1"
+    sha256 "44c14588c1b9b8fcc67326c4119e8d42c8a069670630eab5a1522daac2219708"
+  end
+
+  patch do
+    url "https://github.com/nest/nest-simulator/commit/5cfa2a77a060eae0f90182c56c42bdc8549dd19c.patch?full_index=1"
+    sha256 "69e4d642555a7abefc65dc287d48715ffb42faac72e1ff3dfb880fda58564045"
+  end
+
+  option "with-python", "Build Python3 bindings (PyNEST) instead of Python2 bindings."
```
Comment (fxcoudert) [inline Formula/nest.rb]: Let's choose a python version, and build it with no options. I recommend moving to python3.

File: Formula/nest.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  depends_on "libtool" => :run
+  depends_on "gcc" # scipy, numpy require fortran
+  depends_on "gsl"
+  depends_on "readline"
+
+  # Any Python >= 2.7 < 3.x is okay (either from macOS or brewed)
+  # core only supports macOS and https://docs.brew.sh/Python-for-Formula-Authors.html#bindings
+  depends_on "python" => :optional
+
+  resource "Cython" do
```
Comment (fxcoudert) [inline Formula/nest.rb]: We have formulas for python, numpy, scipy. Can we use those?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
Thanks @tammoippen, I am going to close this pull request for now, but we'll be very happy to consider an updated PR that addresses the review comments and uses the next version of nest!

**Pull request closed without merging**